### PR TITLE
refactor(node-pool) drop stamp usage 

### LINKED ISSUE #1521: Drop Stamp usage from node-pool module

### PR DESCRIPTION
**Pull request by Kalovelo**
closes #1521

This PR drops stamp usage on node-pool and also assigns its usage as an entity instead of composition. For example, in order to access `selectedNode`, it shall be done now as `this.nodePool.selectedNode` instead of  `this.selectedNode`

### CODE REVIEW

File: docs/guides/migration/11.0.0.md [hunk 1]
```diff
@@ -94,7 +94,7 @@ console.log(tx.decodedEvents)
 
 ## Following contract instance methods are dropped([#1368](https://github.com/aeternity/aepp-sdk-js/pull/1368))
 * `topBlock`
-    * use `aeSdk.api.getTopHeader()` instead
+    * use `aeSdk.nodePool.api.getTopHeader()` instead
```
Comment (subhod-i) [inline docs/guides/migration/11.0.0.md:97]:
```suggestion
    * use `aeSdk.api.getTopHeader()` instead
```

This is the migration guide for the older version. The new version will be v12. Can you create a new migration file `12.0.0.md` and add this information there.

File: src/node-pool/index.ts [hunk 2]
```diff
@@ -19,7 +18,7 @@ interface NodeInfo {
   consensusProtocolVersion: string
 }
 
-interface NodePool {
+interface INodePool {
```
Comment (subhod-i) [inline src/node-pool/index.ts:21]: Why do we need an Interface here? @Kalovelo

File: src/node-pool/index.ts [hunk 3]
```diff
@@ -31,33 +30,25 @@ interface NodePool {
...
 }
 
-class _NodePool implements NodePool {
-  readonly api: Node['api']
+export default class NodePool implements INodePool {
   pool: Map<string, Node>
   selectedNode?: Node
 
-  init ({ nodes = [] }: { nodes: Node[] }): void {
+  public get api (): Node['api'] {
```
Comment (subhod-i) [inline src/node-pool/index.ts:37]: Formatting: constructor first, `api` method next down the line?

**Comment by subhod-i** [comment]
Apart from the comments, LGTM.

**Comment by Kalovelo** [comment]
> Apart from the comments, LGTM.

Thank you for your comments, I will update the PR. LGTM seems to be passing without warnings or errors, am I missing something?

File: docs/guides/migration/11.0.0.md (hunk 1 above)
Comment (davidyuk) [reply docs/guides/migration/11.0.0.md]: Actually, we are using `BREAKING CHANGE` commit message

**Pull request closed without merging**